pages/index.js

```diff
@@ -1,12 +1,30 @@
 import Header from "@/components/Header/Header";
 import TaskList from "@/components/TaskList/TaskList";
+import ShoppingItemForm from "@/components/ShoppingItemForm/ShoppingItemForm";
 
-export default function HomePage() {
+export default function HomePage({
+  currentTasks,
+  onCreateTask,
+  handleAddItem,
+  showForm,
+  setShowForm,
+}) {
   return (
     <>
-      <Header />
+      <Header showForm={showForm} setShowForm={setShowForm} />
       <main>
-        <TaskList />
+        <TaskList
+          currentTasks={currentTasks}
+          onCreateTask={onCreateTask}
+          showForm={showForm}
+          setShowForm={setShowForm}
+        />
```

Comment:
Your TaskList only needs the `currentTasks` as their prop:

```suggestion
        <TaskList
          currentTasks={currentTasks}
        />
```